scipy/cluster/tests/test_vq.py

```diff
@@ -214,6 +214,13 @@ def test_kmeans2_rank1_2(self):
         data1 = data[:, 0]
         kmeans2(data1, 2, iter=1)
 
+    def test_kmeans2_high_dim(self):
+        # test kmeans2 when the number of dimensions exceeds the number
+        # of input points
+        data = np.fromfile(DATAFILE1, sep=", ")
+        data = data.reshape((20, 20))[:10]
+        kmeans2(data, 2)
```

Comment:
There should also be a test checking the produced result has the correct covariance.
IOW, use a large `k`, set `np.random.seed(1234)`, and compare the result from `np.cov(x, rowvar=0)` to `np.cov(data, rowvar=0)`